aspnetcore/host-and-deploy/health-checks.md

```diff
@@ -0,0 +1,614 @@
+
+The sample app includes examples of the scenarios described in this topic. To run the sample app for a given scenario, use the [dotnet run](/dotnet/core/tools/dotnet-run) command from the project's folder in a command shell. See the sample app's *README.md* file and the scenario descriptions in this topic for details on how to use the sample app.
+
+## Prerequisites
+
+Health checks are usually used with an external monitoring service or container orchestrator to check the status of an app. Before building a production health check app, decide on which monitoring system to use. The monitoring system dictates what types of health checks to create and how to configure their endpoints.
+
+Reference the [Microsoft.AspNetCore.App metapackage](xref:fundamentals/metapackage-app) or add a package reference to the [Microsoft.AspNetCore.Diagnostics.HealthChecks](https://www.nuget.org/packages/Microsoft.AspNetCore.Diagnostics.HealthChecks) package.
+
+The sample app provides start-up code to demonstrate health checks for several scenarios. The [database probe](#database-probe) scenario probes the health of a database connection using [BeatPulse](https://github.com/Xabaril/BeatPulse). The [database context](#database-context) scenario probes a database context. To explore the database scenarios using the sample app:
+
+* Create a database and provide its connection string in the *appsettings.json* file of the app.
+* Add a package reference to [HealthChecks.SqlServer](https://www.nuget.org/packages/HealthChecks.SqlServer/).
+
+> [!NOTE]
+> [BeatPulse](https://github.com/Xabaril/BeatPulse) isn't maintained or supported by Microsoft.
+
+Another health check scenario demonstrates how to filter health checks to a management port. The sample app requires you to create a *Properties/launchSettings.json* file that includes the management URL and management port. For more information, see the [Filter by port](#filter-by-port) section.
+
+## Basic health probe
+
+For many apps, a basic health probe configuration that reports the app's availability to process requests (*liveness*) is sufficient to discover the status of the app.
+
+The basic configuration registers health check services and calls the Health Check Middleware to respond at a URL endpoint with a health response. By default, no specific health checks are registered to test any particular dependency or subsystem. The app is considered healthy if it's capable of responding at the health endpoint URL. The default response writer writes the status (`HealthCheckStatus`) as a plaintext response back to the client, indicating either a `HealthCheckResult.Healthy` or `HealthCheckResult.Unhealthy` status.
+
+Register health check services with `AddHealthChecks` in `Startup.ConfigureServices`. Call Health Check Middleware in the app processing pipeline in `Startup.Configure`.
```

Comment:
The statement "Call Health Check Middleware" is really odd.

You don't call a middleware, they **get called** by requests

Reply:
This might be a slightly broader problem. We do often say 'add to the middleware processing pipeline,' but we may have similar language to what I have here in other topics. I've made a note to investigate that separately. I'll update this on the next commit.

Reply:
Just to finish off the point :gun: ... "call middleware," i.e., "*call a middleware extension method when setting up the request processing pipeline*" actually **_isn't_** used anywhere in the docs. *Good!* 😅